ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/translator/ASTConverter.java

```diff
@@ -277,8 +277,8 @@ private ASTNode convert() throws CalciteSemanticException {
     if (where != null) {
       ASTNode cond = where.getCondition().accept(new RexVisitor(schema, false, root.getCluster().getRexBuilder()));
       hiveAST.where = ASTBuilder.where(cond);
-      planMapper.link(cond, where);
       planMapper.link(cond, RelTreeSignature.of(where));
+      planMapper.link(cond, where);
```

Comment:
We link RelTreeSignature first so that we can safely unify multiple filters.

Reply:
cool! :D